### PR TITLE
Transfered functions from ProfileConfig to DirectoryProvider

### PR DESCRIPTION
**Pull request by joyastockwell**
Functionality of ProfileConfig now in DirectoryProvider

### CODE REVIEW

File: src/main/kotlin/persistence/DirectoryProvider.kt [hunk 1]
```diff
@@ -47,4 +49,31 @@ class DirectoryProvider(private val appName: String) : IDirectoryProvider {
...
+        val pathDir = getAppDataDirectory(directoryName)
+        println("path: " + pathDir)
+        return pathDir.path
+    }
+
+    // Creates a file in the private internal storage of the given application
+    fun createFile(path: String, filename: String, fileToStore: File): String {
+        val fileNew = File(path, filename)
+        fileToStore.copyTo(fileNew, true)
+        return fileNew.path
```
Comment (jsarabia) [inline src/main/kotlin/persistence/DirectoryProvider.kt]: return File objects not Strings

File: src/main/kotlin/persistence/DirectoryProvider.kt [hunk 2]
```diff
@@ -47,4 +48,21 @@ class DirectoryProvider(private val appName: String) : IDirectoryProvider {
...
+
+    // Stores the image with name of the user who stored it, followed by the original name of the file
+    fun storeImageDirectory(userMine: User, imageFile: File): File {
+        return createInternalAppDirectory("Images")
+    }
+
+    // Returns path of new directory
+    fun createInternalAppDirectory(directoryName: String): File {
+        val pathDir = getAppDataDirectory(directoryName)
+        return pathDir
```
Comment (mbr4477) [inline src/main/kotlin/persistence/DirectoryProvider.kt]: Do we need this function if it just calls `getAppDataDirectory()`?

File: src/main/kotlin/persistence/DirectoryProvider.kt [hunk 3]
```diff
@@ -47,4 +48,21 @@ class DirectoryProvider(private val appName: String) : IDirectoryProvider {
         file.mkdirs()
         return file
     }
+
+    // Creates a user subdirectory and stores their hash info
+    fun getAudioDirectory(): File {
```
Comment (mbr4477) [inline src/main/kotlin/persistence/DirectoryProvider.kt]: Maybe change this to `getUserProfileAudioDirectory()` to better explain what the directory is

File: src/main/kotlin/persistence/DirectoryProvider.kt [hunk 4]
```diff
@@ -47,4 +48,21 @@ class DirectoryProvider(private val appName: String) : IDirectoryProvider {
...
+        return createInternalAppDirectory("Audio")
```
Comment (mbr4477) [inline src/main/kotlin/persistence/DirectoryProvider.kt]: The directory should probably be something more like `getAppDataDirectory("users${separator}audio")` for better organization

File: src/main/kotlin/persistence/DirectoryProvider.kt [hunk 5]
```diff
@@ -47,4 +48,21 @@ class DirectoryProvider(private val appName: String) : IDirectoryProvider {
...
+    }
+
+    // Stores the image with name of the user who stored it, followed by the original name of the file
+    fun storeImageDirectory(userMine: User, imageFile: File): File {
```
Comment (mbr4477) [inline src/main/kotlin/persistence/DirectoryProvider.kt]: Maybe change to `getUserProfileImageDirectory()` to match the pattern of the other methods

File: src/main/kotlin/persistence/DirectoryProvider.kt [hunk 6]
```diff
@@ -47,4 +48,21 @@ class DirectoryProvider(private val appName: String) : IDirectoryProvider {
...
+        return createInternalAppDirectory("Images")
```
Comment (mbr4477) [inline src/main/kotlin/persistence/DirectoryProvider.kt]: Same as audio function, change to something like `getAppDataDirectory("users${separator}images")`

**Comment by joyastockwell** [comment]
Made the changes

**Review by jsarabia — APPROVED** [review]

**Pull request merged**